lib/parsers.js

```diff
@@ -53,123 +53,105 @@ module.exports.parseError = function parseError(err, kwargs, cb) {
   });
 };
 
-module.exports.parseRequest = function parseRequest(req, kwargs) {
-  kwargs = kwargs || {};
+module.exports.parseRequest = function parseRequest(req, parseUser) {
+  var kwargs = {};
 
   // headers:
-  //
-  //   node: req.headers
-  //   express: req.headers
+  //   node, express: req.headers
   //   koa: req.header
-  //
   var headers = req.headers || req.header || {};
 
   // method:
-  //
-  //   node: req.method
-  //   express: req.method
-  //   koa: req.method
-  //
+  //   node, express, koa: req.method
   var method = req.method;
 
   // host:
-  //
-  //   node: req.headers.host
   //   express: req.hostname in > 4 and req.host in < 4
   //   koa: req.host
-  //
+  //   node: req.headers.host
   var host = req.hostname || req.host || headers.host || '<no host>';
 
   // protocol:
-  //
   //   node: <n/a>
-  //   express: req.protocol
-  //   koa: req.protocol
-  //
+  //   express, koa: req.protocol
   var protocol = req.protocol === 'https' || req.secure || (req.socket || {}).encrypted ? 'https' : 'http';
 
   // url (including path and query string):
-  //
-  //   node: req.originalUrl
-  //   express: req.originalUrl
+  //   node, express: req.originalUrl
   //   koa: req.url
-  //
   var originalUrl = req.originalUrl || req.url;
 
   // absolute url
-  var url = protocol + '://' + host + originalUrl;
+  var absoluteUrl = protocol + '://' + host + originalUrl;
 
-  // query string
-  //
+  // query string:
   //   node: req.url (raw)
-  //   express: req.query
-  //   koa: req.query
-  //
+  //   express, koa: req.query
   var query = req.query || urlParser.parse(originalUrl || '', true).query;
 
   // cookies:
-  //
-  //   node: req.headers.cookie
-  //   express: req.headers.cookie
-  //   koa: req.headers.cookie
-  //
+  //   node, express, koa: req.headers.cookie
   var cookies = cookie.parse(headers.cookie || '');
 
   // body data:
-  //
-  //   node: req.body
-  //   express: req.body
-  //   koa: req.body
-  //
+  //   node, express, koa: req.body
   var data = req.body;
   if (['GET', 'HEAD'].indexOf(method) === -1) {
     if (typeof data === 'undefined') {
       data = '<unavailable>';
     }
   }
 
-  if (data && {}.toString.call(data) !== '[object String]') {
+  if (data && typeof data !== 'string' && {}.toString.call(data) !== '[object String]') {
```

Comment:
The `{}.toString.call` check is really checking two cases without being obvious about it, and I sort of prefer using `typeof` as much as possible and avoiding checking object toString types when we don't need to be. This better delineates/escalates through the cases we care about IMO (and `typeof` is a cheaper check in the common case, not that we care about a trivial perf diff in a once-per-error check).